docs/hugo/content/docs/authentication.md

```diff
@@ -0,0 +1,95 @@
+# Authentication in Azure Service Operator v2
+
+Azure Service Operator supports two different styles of authentication today.
+
+1. Service Principal
+2. aad-pod-identity authentication (managed identity)
+
+## Service Principal
+
+### Prerequisites
+1. An existing Azure Service Principal.
+
+To use Service Principal authentication, specify an `aso-controller-settings` secret with `AZURE_CLIENT_SECRET` set.
+`AZURE_CLIENT_ID` must be set to the Service Principal client ID.
+`AZURE_CLIENT_SECRET` must be set to the Service Principal client secret.
```

Comment:
```suggestion
To use Service Principal authentication, specify an `aso-controller-settings` secret with `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET` set.

* `AZURE_CLIENT_ID` must be set to the Service Principal client ID.
* `AZURE_CLIENT_SECRET` must be set to the Service Principal client secret.
```

Maybe give an example of each, so that people can recognize when they've found the right value to include?

Reply:
Giving an example here is a bit awkward because both fields are pretty inscrutable (CLIENT_ID is a GUID and the password is some black-box base64'ed string that Azure gives to them when they create the SP). What I've done is call out that client ID is a GUID and also link to the documentation in the Azure CLI that shows how to create an SP (which will return these values).